app/app_mobile_and.go

```diff
@@ -25,8 +25,11 @@ func (app *fyneApp) OpenURL(url *url.URL) error {
 }
 
 func rootConfigDir() string {
-	homeDir := "/data" //, _ := os.UserHomeDir()
+	filesDir := os.Getenv("FILESDIR")
+	if filesDir == "" {
+		log.Println("FILESDIR env was not set by android native code")
+		return "/data/data" // probably won't work, but we can't make a better guess
```

Comment:
This path will almost certainly not be readable or writable, but if you can get the package name instead return "data/data/<package>/files/"?

Reply:
Indeed, that is what FILESDIR is providing. The fallback is for when it was not possible to look up the package info (or if the packager l didn’t yet support that feature) so we can’t look into the android runtime when falling back...

Reply:
Where does FILESDIR get set?

Reply:
https://github.com/fyne-io/fyne/commit/e8ddd407ac01dbd772d84039c2183f6a091aa64f#diff-9930725eda019b4f278d281d5d77ab2fR89

Reply:
Ah kk, so it is getting set to internalDataPath and if that fails there will be a log entry. Is returning /data/data any better than returning an empty string?

Reply:
Well, as it gets appended to its a choice between desperately hoping that /data/data is writeable vs hoping that the file system root is writeable...

Reply:
I think you're out of luck either way - but it doesn't matter too much because if a dev encounters an unwritable error they need to realize the issue is not with the unwritable error and instead they need to figure out why the NDK did not set the path, which your log helps them realize